packages/test262-runner/tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "../../tsconfig.eslint-base.json",
+  "compilerOptions": {
+    "allowJs": true
+  },
```

Comment:
For this PR, no change suggested.

But I've been meaning to ask about this. Some packages have a `tsconfig.json` without this section. Some have this section. And some have it with an additional 
```json
    "maxNodeModuleJsDepth": 1, // or 2
```
At #2042 I found I had to add one to @endo/pass-style to get the types to show up in my IDE as expected. I gave it a max depth of 1, but not for any particular reason. Can we be uniform? Is there a default config for new packages?

Reply:
I’m alarmed that this was necessary within Endo. My read of how this feature works is that TSC literally counts occurrences of the term `node_modules` in the canonical path to the module, not the shortest dependency path. Within the Endo repository, the physical locations of each package contain _no_ path components of `node_modules` so TSC should reach indefinitely. This was why we didn’t encounter the need for this feature until packages started moving out of Agoric SDK, increasing the depth.

But, we may have just been lucky until now. I find it alarming that we would ever need this for local development in Endo.

Reply:
@turadg , do you understand what the effects are of including vs omitting a `"compilerOptions":` section in tsconfig.js files? Are you surprised that without them I was at least missing type info in my vscode IDE?

Reply:
The effect of `compilerOptions` from a tsconfig is to use the TS defaults, or those inherited from another config.

The effect of each option is described in https://www.typescriptlang.org/tsconfig#compilerOptions

In Endo, the options for "build" are different from normal dev. The dev tsconfig in every package inherits from these this file:
- https://github.com/endojs/endo/blob/75bd2d2f71038bc720af7d927b0052041f07f199/tsconfig.eslint-base.json#L2-L12

The options used in various packages:
- `checkJs` - whether to check all `.js` files or just the ones with `// @ts-check`
- `allowSyntheticDefaultImports` - whether to allow `import x from y` when the module doesn't have a default export (like CJS)
- `noImplicitAny` - a strictness check to prevent unannotated values from silently being `any`.
- [maxNodeModuleJsDepth](https://www.typescriptlang.org/tsconfig#maxNodeModuleJsDepth) - The maximum dependency depth to search under node_modules and load JavaScript files.

> Are you surprised that without them I was at least missing type info in my vscode IDE?

Not surprising that that state was encountered. When refactoring packages, the search depth may have to be increased because we don't build `.d.ts` files in development.

> Can we be uniform? Is there a default config for new packages?

The default is zero (i.e. when not specified). The only way to be uniform is to adopt the smallest number that works for all packages, but that would make some packages slower and possibly introduce new errors as they resolve something new that then needs more coverage to continue.

Note that this is only an issue while working in an IDE. The typedoc build in CI ensures that the types of the entire repo resolve sanely.

If we want to also ensure that type coverage never falls on a per-package basis, we can add a check to CI that uses compares its `typeCoverage` value in its package.json with the coverage after the PR changes.